File: services/auth-service/src/main/java/com/medmatch/auth/service/AuthService.java
package com.medmatch.auth.service;

import com.medmatch.auth.dto.AuthResponse;
import com.medmatch.auth.dto.LoginRequest;
import com.medmatch.auth.dto.RegisterRequest;
import com.medmatch.auth.dto.UserResponse;

public interface AuthService {

    UserResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);
}

File: frontend/medmatch-ui/src/features/auth/api/auth.service.ts
import { authApi } from "@/api/axios";
import type {
  LoginRequest,
  LoginResponse,
} from "@/features/auth/types/auth";

class AuthService {
  async login(
    request: LoginRequest
  ): Promise<LoginResponse> {
    const response = await authApi.post<LoginResponse>(
      "/auth/login",
      request
    );

    return response.data;
  }
}

export const authService = new AuthService();